### PR TITLE
Code cleanup: use IndexOf(char) overload

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

- Use IndexOf(char) overload instead of IndexOf(string) to benefit from Ordinal search
- Replace IndexOf with Contains to get more understandable code
- Use StringComparison.Ordinal for ASCII chars

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by iSazonov** [comment]
@daxian-dbw @adityapatwardhan Could you please review?

File: src/Microsoft.PowerShell.Security/security/CertificateProvider.cs [hunk 1]
```diff
@@ -3204,8 +3204,8 @@ public DnsNameProperty(X509Certificate2 cert)
             // extract DNS name from subject distinguish name
             // if it exists and does not contain a comma
             // a comma, indicates it is not a DNS name
-            if (cert.Subject.StartsWith(distinguishedNamePrefix, System.StringComparison.InvariantCultureIgnoreCase) &&
-                cert.Subject.IndexOf(",", System.StringComparison.InvariantCulture) == -1)
+            if (cert.Subject.StartsWith(distinguishedNamePrefix, System.StringComparison.OrdinalIgnoreCase) &&
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Security/security/CertificateProvider.cs]: Can we move this to another PR?  This looks like a bug fix

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
See comment

File: src/Microsoft.PowerShell.Security/security/CertificateProvider.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Security/security/CertificateProvider.cs]: Moved to #9987

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
@TravisEz13 Please update your review.

**Review by TravisEz13 — APPROVED** [review]
